### PR TITLE
EVG-13844: Render 5 commit charts from query data

### PR DESCRIPTION
**Pull request by ziyielenachen**
[EVG-13844](https://jira.mongodb.org/browse/EVG-13844)
This is branched off of [EVG-14905](https://github.com/evergreen-ci/spruce/pull/807), so they share most of the files changed. Main changes for this branch are in `Commits.tsx` and `CommitsWrapper.tsx`.

### Description 
* New waterfall page now uses queried data from mainlineCommits to display active commit charts and chart labels 

### Screenshots
Absolute view:
Screen Shot 2021-07-16 at 4 59 03 PM

Percentage view:
Screen Shot 2021-07-16 at 4 59 10 PM

### Testing 
Added Storybook test ProjectHealth that displays active commit charts and chart labels

### CODE REVIEW

File: src/pages/commits/CommitsWrapper.tsx [hunk 1]
```diff
@@ -29,9 +33,33 @@ export const CommitsWrapper: React.FC<Props> = ({
     return <StyledSkeleton active title={false} paragraph={{ rows: 6 }} />;
   }
   if (!isLoading && versions?.length !== 0) {
+    const IdToTaskStatsGroupedByColor = getAllTaskStatsGroupedByColor(versions);
```
Comment (arjunpatel100) [inline src/pages/commits/CommitsWrapper.tsx:36]: nit: lowercase first letter

File: src/pages/commits/CommitsWrapper.tsx [hunk 2]
```diff
@@ -29,9 +33,33 @@ export const CommitsWrapper: React.FC<Props> = ({
...
+                />
+                <CommitChartLabel
+                  githash={item.version.revision.substring(0, 5)}
+                  createTime={item.version.createTime}
+                  author={item.version.author}
+                  message={item.version.message}
+                />
+              </ActiveCommitWrapper>
+            ) : null
+          )}
```
Comment (arjunpatel100) [inline src/pages/commits/CommitsWrapper.tsx:61]:
```suggestion
          {versions.map(({version})) =>
            item.version ? (
              <ActiveCommitWrapper key={version.id}>
                <CommitChart
                  groupedTaskStats={
                    IdToTaskStatsGroupedByColor[version.id].stats
                  }
                  total={IdToTaskStatsGroupedByColor[version.id].total}
                  max={max}
                  chartType={chartType}
                />
                <CommitChartLabel
                  githash={version.revision.substring(0, 5)}
                  createTime={version.createTime}
                  author={version.author}
                  message={version.message}
                />
              </ActiveCommitWrapper>
            ) : null
          )}
```

**Review by arjunpatel100 — APPROVED** [review]

File: src/pages/commits/ActiveCommits/CommitChart.tsx [hunk 3]
```diff
@@ -35,28 +29,20 @@ export const CommitChart: React.FC<Props> = ({
...
-        />
-      ) : null
-    )}
+    {groupedTaskStats.map((colorCount) => (
+      <Bar
+        key={colorCount.color}
+        height={calculateHeight(colorCount.count, max, total, chartType)}
+        color={colorCount.color}
+      />
+    ))}
```
Comment (arjunpatel100) [inline src/pages/commits/ActiveCommits/CommitChart.tsx:38]:
```suggestion
    {groupedTaskStats.map(({color, count})) => (
      <Bar
        key={color}
        height={calculateHeight(count, max, total, chartType)}
        color={color}
      />
    ))}
```

File: src/utils/statuses/groupStatusesByColor.test.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        count: 6,
+        statuses: [TaskStatus.Dispatched, TaskStatus.WillRun],
+        color: mapTaskStatusToColor[TaskStatus.Dispatched],
+      },
+    ],
+    max: 8,
+    total: 21,
+  });
+});
+test("Returns the overall maximum and toatl", () => {
```
Comment (arjunpatel100) [inline src/utils/statuses/groupStatusesByColor.test.ts:70]:
```suggestion
test("Returns the overall maximum and total", () => {
```

File: src/utils/statuses/groupStatusesByColor.test.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
+import { mapTaskStatusToColor } from "constants/task";
+import { TaskStatus } from "types/task";
+import { groupStatusesByColor } from "./groupStatusesByColor";
+
+test("Seperates statuses into groups based on the color of the status", () => {
```
Comment (arjunpatel100) [inline src/utils/statuses/groupStatusesByColor.test.ts:5]:
```suggestion
test("Separates statuses into groups based on the color of the status", () => {
```

**Review by arjunpatel100 — APPROVED** [review]

File: src/pages/commits/CommitsWrapper.tsx [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
 import { MainlineCommitsQuery } from "gql/generated/types";
-import { ChartToggle } from "pages/commits/commitChart/ChartToggle";
-import { Grid } from "pages/commits/commitChart/Grid";
-import { GroupedResult } from "pages/commits/commitChart/utils";
+import { CommitChart } from "pages/commits/ActiveCommits/CommitChart";
+import { CommitChartLabel } from "pages/commits/ActiveCommits/CommitChartLabel";
+import {
+  getAllTaskStatsGroupedByColor,
+  findMaxGroupedTaskStats,
+} from "pages/commits/ActiveCommits/utils";
```
Comment (khelif96) [inline src/pages/commits/CommitsWrapper.tsx:11]: `./ActiveCommits/...`

**Review by khelif96 — APPROVED** [review]
Commits and CommitsWrapper.tsx lgtm

File: src/utils/statuses/groupStatusesByColor.test.ts (hunk 4 above)
Comment (ziyielenachen) [reply src/utils/statuses/groupStatusesByColor.test.ts:70]: Good eyes!!

**Pull request merged**